src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,414 @@
+                                                             @Nullable FailureCollector collector,
+                                                             @Nullable String recordPrefix) {
+    com.google.cloud.dataplex.v1.Schema.Type standardType = field.getType();
+    switch (standardType) {
+      case FLOAT:
+      case DOUBLE:
+        // float is a float64, so corresponding type becomes double
+        return Schema.of(Schema.Type.DOUBLE);
+      case BOOLEAN:
+        return Schema.of(Schema.Type.BOOLEAN);
+      case BYTE:
+      case INT16:
+      case INT32:
+      case INT64:
+        return Schema.of(Schema.Type.LONG);
+      case STRING:
+        return Schema.of(Schema.Type.STRING);
+      case BINARY:
+        return Schema.of(Schema.Type.BYTES);
+      case TIME:
+        return Schema.of(Schema.LogicalType.TIME_MICROS);
+      case DATE:
+        return Schema.of(Schema.LogicalType.DATE);
+      case TIMESTAMP:
+        return Schema.of(Schema.LogicalType.TIMESTAMP_MICROS);
+      case DECIMAL:
+        // bigquery has Numeric.PRECISION digits of precision and Numeric.SCALE digits of scale for NUMERIC.
+        // https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
+        return Schema.decimalOf(BigQueryTypeSize.Numeric.PRECISION, BigQueryTypeSize.Numeric.SCALE);
+      case NULL:
+        return Schema.of(Schema.Type.NULL);
+      default:
+        String error =
+          String.format("Entity column '%s' is of unsupported type '%s'.", field.getName(), standardType.name());
+        String action = String.format("Supported column types are: %s.",
+          BigQueryUtil.BQ_TYPE_MAP.keySet().stream().map(t -> t.getStandardType().name())
+            .collect(Collectors.joining(", ")));
+        if (collector != null) {
+          collector.addFailure(error, action);
+        } else {
+          throw new RuntimeException(error + action);
+        }
+        return null;
+    }
+  }
+
+  /**
+   * @param conf contains all the configuration properties.
+   * @throws IOException
+   */
+  public static void getJobCompletion(Configuration conf) throws IOException {
+    GoogleCredentials googleCredentials = getCredentialsFromConfiguration(conf);
+    String projectID = conf.get(DataplexConstants.DATAPLEX_PROJECT_ID);
+    String location = conf.get(DataplexConstants.DATAPLEX_LOCATION);
+    String lake = conf.get(DataplexConstants.DATAPLEX_LAKE);
+    String taskId = conf.get(DataplexConstants.DATAPLEX_TASK_ID);
+    try (DataplexServiceClient dataplexServiceClient = DataplexUtil.getDataplexServiceClient(googleCredentials)) {
+      DataplexServiceClient.ListJobsPagedResponse
+        jobList = dataplexServiceClient.listJobs(TaskName.newBuilder().setProject(projectID).setLake(lake).
+        setLocation(location).setTask(taskId).build());
+      Job dataplexJob = jobList.iterateAll().iterator().next();
+      try {
+        Awaitility.await()
+          .atMost(30, TimeUnit.MINUTES)
```

Comment:
30 minutes seems like quite a long time. How did we arrive at this number?

Reply:
We have tested for 5 million records and it is taking around 10 min in creating dataplex task and then dataplex job and finally giving the output. For dataplex we can have millions of data, for that reason I set it to 30 min.